packages/cli/generators/repository/index.js

```diff
@@ -0,0 +1,393 @@
+    super(args, opts);
+  }
+
+  async _getModelFields(modelName) {
+    let retArray = [];
+    let fileContent = '';
+
+    let modelFile = path.join(
+      this.artifactInfo.modelDir,
+      modelName.toLowerCase() + '.model.ts',
+    );
+
+    try {
+      fileContent = this.fs.read(modelFile, {});
+    } catch (err) {
+      debug(`${ERROR_READING_FILE} ${modelFile}: ${err.message}`);
+      return this.exit(err);
+    }
+
+    const ast = tsquery.ast(fileContent);
+    const nodes = tsquery(ast, ':matches([kindName="PropertyDeclaration"])');
+
+    for (let x of nodes) {
+      retArray.push(x.name.escapedText);
+    }
+
+    return retArray;
+  }
+
+  async _getModelIdType(modelName) {
+    let fileContent = '';
+
+    let modelFile = path.join(
+      this.artifactInfo.modelDir,
+      modelName.toLowerCase() + '.model.ts',
+    );
+
+    try {
+      fileContent = this.fs.read(modelFile, {});
+    } catch (err) {
+      debug(`${ERROR_READING_FILE} ${modelFile}: ${err.message}`);
+      return this.exit(err);
+    }
+
+    const ast = tsquery.ast(fileContent);
+    const nodes = tsquery(ast, 'TrueKeyword');
+    let retVal = 'none';
+
+    for (let prop of nodes) {
+      if (prop.parent.name.parent.text === 'id: true') {
+        /**
+         * prop.parent.name.parent.parent.text returns something like this, and
+         * it is a malformed json.
+         *  {
+         *   type: 'string',
+         *   id: true,
+         *   default: name,
+         *  }
+         */
+        let a = new String(prop.parent.name.parent.parent.text).replace(
+          /['{}\n]/g,
+          '',
+        );
+        let b = a.split(',');
+        for (let c of b) {
+          if (c.includes('type')) {
+            const tempVal = c.replace('type:', '').trim();
+            // let's make sure we parsed the datatype correctly
+            if (VALID_PROPERTY_ID_TYPES.includes(tempVal)) {
+              retVal = tempVal;
+            }
+            break;
+          }
+        }
+      }
+    }
+
+    return retVal;
+  }
+
+  /**
+   * load the connectors available and check if the basedModel matches any
+   * connectorType supplied for the given connector name
+   * @param {string} connectorType single or a comma separated string array
+   */
+  async _isConnectorOfType(connectorType, dataSourceClassName) {
+    debug(`calling isConnectorType ${connectorType}`);
+    let jsonFileContent = '';
+    let result = false;
+
+    let datasourceJSONFile = path.join(
+      this.artifactInfo.datasourcesDir,
+      dataSourceClassName
+        .replace('Datasource', '.datasource.json')
+        .toLowerCase(),
+    );
+
+    try {
+      jsonFileContent = this.fs.readJSON(datasourceJSONFile, {});
+    } catch (err) {
+      debug(`${ERROR_READING_FILE} ${datasourceJSONFile}: ${err.message}`);
+      return this.exit(err);
+    }
+
+    for (let connector of Object.values(connectors)) {
+      if (
+        jsonFileContent.connector === connector.name &&
```

Comment:
I believe we allow two options how to specify a connector used by a datasource, at least that's the case of LB 3.x CLI and both LB 3.x/4.x runtime:

1. Using the full name, e.g. `loopback-connector-mysql`.
2. Using a short name without `loopback-connector-` prefix, i.e. `mysql`.

Reply:
So we should ask for both, short as of now and the longer version right?

Reply:
The prompt for a connector to use is handled by `lb4 datasource` command, no need to change that.

I am saying that you should improve this check to correctly handle both long and short form of the `connector` property specified in the datasource configuration.

E.g.

```ts
const connector = jsonFileContent.connector;

if (
  connector === connector.name ||
  `loopback-connector-${connector}` === connector.name
  // ...
) {
}
```

Also please refactor this multi-line `if` condition to fit into a single line, see [Indentation of multi-line expressions in if](https://loopback.io/doc/en/contrib/style-guide.html#indentation-of-multi-line-expressions-in-if), the example called "Best".